Add current sensor calibration documentation (#95)
Severson-Group/docs.amdc.dev: source/getting-started/control-with-amdc/current-sensor-cal/index.md

1. First, note the sensor reading when there is no current flowing through phase U.
1. Next, apply a differential open loop voltage on phase U to cause some some current to flow through phase U. The value of voltage is left to the discretion of the user based on the system nominal ratings.
1. Note down the sensor reading as well as the reference current flowing through phase U cable.
1. Progressively increase the applied voltage and note down the readings. It is recommended to go up until the rated value of the current is flowing through phase U.
1. Tabulate the measurements as shown in `exp_data.csv` file [here](./resources/exp_data.csv).
1. Fit a linear expression of the form $\text{Reading [V]} = \text{Gain [V/A]} \times \text{Current [A]} + \text{Offset [V]}$ to the obtained measurements. A [Jupyter notebook](./resources/current_sensor_calibration.ipynb) is provided for this purpose.
1. For the data presented, the fit is shown in the below plot.
1. Now the obtained gain and offset can be used in the control code to convert the sensor reading into the actual current measurement.
1. Repeat the exercise for phases U and V of the system

> <img src="./resources/fit.svg" width="50%" align="center"/>

Reviewer: @anirudhupadhyaya -- it is odd to me that the data goes different amounts negative and positive (i.e., it runs -15A to +10A). Can you truncate your data so that you just show it over a range of approx. -10A to approx. +10A?

1. Tabulate the measurements as shown in this example [`exp_data.csv` file](./resources/exp_data.csv).
1. Fit a linear expression of the form $\text{Reading [V]} = \text{Gain [V/A]} \times \text{Current [A]} + \text{Offset [V]}$ to the obtained measurements. This [example Jupyter notebook](./resources/current_sensor_calibration.ipynb) is provided to illustrate the process.
1. Repeat the exercise for the remaining phases of the system.

An example of the results are shown in the plot below. The obtained gain and offset can be used directly in the control code to convert the sensor reading into the actual current measurement.

<img src="./resources/fit.svg" width="50%" align="center"/>

```{tip}
Be sure to conduct the calibration process over the full range of current data (both positive and negative current) to account for any variation in the current sensor reading due to directionality of current.
```
## Use of Calibration Data

The below codeblock can be utilized by the user to convert between raw measurements from the sensor and the actual currents.

```C
#define INV_GAIN (1.0/0.621) // Inverse of gain obtained from curve fit (1/0.621)

> double offset 4.739; // [V], offset from curve fit. This is a variable so that the drive can remeasure and adjust the offset at startup 

Reviewer: ```suggestion
double offset = 4.739; // [V], offset from curve fit. This is a variable so that the drive can remeasure and adjust the offset at startup 
```

> #define OFFSET 4.739 // Offset from curve fit. Note: This would have to be a variable in real implementation, since it is preferred to re-calibrate the offset during boot-up

Reviewer: ```suggestion
double offset 4.739 // [V], offset from curve fit. This is a variable so that the drive can remeasure and adjust the offset at startup 
```

double current_measurement; // Actual current measurement, to be used in control algorithm

> current_measurement = (sensor_reading - OFFSET)*INV_GAIN;  // sensor_reading is the raw measurement and needs to be obtained by the user

Reviewer: ```suggestion
current_measurement = (sensor_reading - offset)*INV_GAIN;  // sensor_reading is the raw measurement and needs to be obtained by the user
```

1. Record the drive's sensor reading as well as the reference sensor's reading of the current flowing through the phase U cable.
1. Progressively increase the phase U current and record the readings. Do this over the full range of rated current, both positive and negative. 
1. Tabulate the measurements as shown in this example [`exp_data.csv` file](./resources/exp_data.csv).
1. Fit a linear expression of the form $\text{Reading [V]} = \text{Gain [V/A]} \times \text{Current [A]} + \text{Offset [V]}$ to the obtained measurements. This [example Jupyter notebook](./resources/current_sensor_calibration.ipynb) is provided to illustrate the process.
1. Repeat the exercise for the remaining phases of the system.

An example of the results are shown in the plot below. The obtained gain and offset can be used directly in the control code to convert the sensor reading into the actual current measurement.

<img src="./resources/fit.svg" width="50%" align="center"/>

```{tip}
Be sure to conduct the calibration process over the full range of current data (both positive and negative current) to account for any variation in the current sensor reading due to directionality of current.
```

> ## Use of Calibration data

Reviewer 2: Capitalize `Data` to match the rest of the title

The below codeblock can be utilized by the user to convert between raw measurements from the sensor and the actual currents.

```C
#define GAIN 0.621 // Gain from curve fit
#define OFFSET 4.739 // Offset from curve fit

double current_measurement; // Actual current measurement, to be used in control algorithm

current_measurement = (sensor_reading - OFFSET)/GAIN;  // sensor_reading is the raw measurement and needs to be obtained by the user

> ```

Reviewer 2: Hmm. Is this really how you implement this in your code? I would expect to see a multiply operation used (typically faster), so the inverse operation would be computed at compile time in the define statement, e.g., `(1.0 / 0.621)` or something like that.

Reviewer 2: Also, a real implementation would be a bit more "complex" since usually we want to recalibrate during boot-up, so having constants as defines does not work... I do not think you need to give the reader all possible code they would want, but can you add a comment explaining this?

Author: I have added `tip`, let me know if this does the job...

Author: I have added a comment alongside `OFFSET` define

Reviewer: I suggest changing `offset` to be a global variable and update the comment. See my suggestion in my review.
